src/main/java/org/opensrp/web/rest/PlanResource.java

```diff
@@ -217,10 +216,10 @@ public void setLocationService(PhysicalLocationService locationService) {
 	 */
 	@RequestMapping(value = "/findLocationNames/{planIdentifier}", method = RequestMethod.GET, produces = {
 	        MediaType.APPLICATION_JSON_VALUE })
-	public ResponseEntity<List<LocationDetail>> findLocationDetailsByPlanId(
```

Comment:
Why is this being changed on this PR

Reply:
A test was failing - not sure how it passed on the main branch. 

The issue here is that the test expects null fields to be ignored but Spring's serialization doesn't ignore them - so I modified it to use gson.

Reply:
I think the main reason for now using domain objects instead  of string is also to have schema documentation in swagger. Why not change the test instead of changing implementation here

Reply:
Most of the other methods use gson serialization in this class. Also, the test is probably a reflection of what the code was supposed to do, so modifying that instead of the code did not feel right.

Reply:
The test is passing in v2.1 and master branches

Reply:
This https://travis-ci.org/github/OpenSRP/opensrp-server-web/builds/693776168 shows it was failing after the merge.

Reply:
It was also failing locally.